src/wallet/wallet.cpp

```diff
@@ -1841,9 +1845,20 @@ CWallet::ScanResult CWallet::ScanForWalletTransactions(const uint256& start_bloc
             WalletLogPrintf("Still rescanning. At block %d. Progress=%f\n", block_height, progress_current);
         }
 
+        bool skip_block{false};
+        if (fast_rescan_filter) {
+            fast_rescan_filter->UpdateIfNeeded();
+            auto matches_block{fast_rescan_filter->MatchesBlock(block_hash)};
+            if (matches_block.has_value() && !*matches_block) {
+                result.last_scanned_block = block_hash;
+                result.last_scanned_height = block_height;
+                skip_block = true;
+            }
+        }
+
         // Read block data
         CBlock block;
-        chain().findBlock(block_hash, FoundBlock().data(block));
+        if (!skip_block) chain().findBlock(block_hash, FoundBlock().data(block));
```

Comment:
In 73964bc1590f0ba1113ecda7895744bd5f870ee5 "wallet: take use of `FastWalletRescanFilter`"

This could be inside of the `if (!skip_block)` below, and thus avoid having to construct a `CBlock` when we aren't going to use it.